include/fiction/algorithms/properties/check_planarity.hpp

```diff
@@ -0,0 +1,104 @@
+//
+
+#ifndef FICTION_CHECK_PLANARITY_HPP
+#define FICTION_CHECK_PLANARITY_HPP
+
+#include "fiction/algorithms/network_transformation/network_balancing.hpp"
+
+#include <mockturtle/traits.hpp>
+
+#include <algorithm>
+#include <cstdint>
+#include <stdexcept>
+
+namespace fiction
+{
+
+template <typename Ntk>
+class check_planarity_impl
+{
+  public:
+    check_planarity_impl(const Ntk& ntk) : ntk(ntk) {}
+
+    /**
+     * @brief Checks if a given network is planar.
+     *
+     * This function checks if the network represented by the variable `ntk` is planar.
+     * The network is planar if for any edge with starting point m and endpoint n (represented by the node ranks), there
+     * is never another edge with starting point m_ > m and endpoint n_ < n, or vice versa. When iterating through
+     * the ranks of one level, the endpoints are always increasing. Therefore, only the starting points need to be
+     * checked. Thus, the highest connected starting point in the fan-in gives a border m_max for every subsequent edge.
+     *
+     * @return `true` if the network is planar, `false` otherwise.
+     */
+    bool run()
+    {
+        bool return_false = false;
+        for (uint32_t r = 1; r < ntk.depth() + 1; r++)
+        {
+            uint32_t bound = 0;
+            ntk.foreach_node_in_rank(r,
+                                     [this, &bound, &return_false](const auto& n)
+                                     {
+                                         uint32_t new_bound = bound;
+                                         ntk.foreach_fanin(n,
+                                                           [this, &bound, &new_bound, &return_false](const auto& fi)
+                                                           {
+                                                               const auto fi_n = ntk.get_node(fi);
+                                                               if (ntk.rank_position(fi_n) < bound)
+                                                               {
+                                                                   return_false = true;
+                                                               }
+                                                               new_bound = std::max(new_bound, ntk.rank_position(fi_n));
+                                                           });
+                                         if (return_false)
+                                             return;
+                                         bound = new_bound;
+                                     });
+            if (return_false)
+                return false;
+        }
+        return true;
+    }
+
+  private:
+    const Ntk ntk;
+};
+
+/**
+ * Checks if a logic network is planar. To perform this check, the network must have ranks assigned.
+ *
+ * If the network is not balanced, an exception is thrown. To balance the network, insert buffers to divide multi-level
+ * edges.
```

Comment:
Why can only balanced networks be checked for planarity? Because you check level by level? Is that a true planarity check then? Should we reflect this in the function name and docstring? Something like `check_rank_planarity` or `check_rank_by_rank_planarity` (just ideas after thinking about it for two seconds; you might come up with something better).